2009-07-24  Adrien DELLE CAVE <decryptus@proformatique.com>

	pf_sip_mwi_for_thomson_st2030s.patch
	$Revision: 6692 $
	$Date: 2009-07-22 19:03:42 +0200 (Wed, 22 Jul 2009) $

	Correct a misuse of strcasecmp for SIP Notification.

	Replaced phone domain by Asterisk domain in Call-ID of the MWI subscription.
	Special fix for Thomson ST2022S and ST2030S.

Index: asterisk-1.4.26/channels/chan_sip.c
===================================================================
--- asterisk-1.4.26.orig/channels/chan_sip.c	2009-07-24 10:03:07.000000000 +0200
+++ asterisk-1.4.26/channels/chan_sip.c		2009-07-24 10:06:54.000000000 +0200
@@ -16690,6 +16690,18 @@
 	if (peer->mwipvt) {
 		/* Base message on subscription */
 		p = peer->mwipvt;
+
+		if (!strncmp(peer->useragent, "THOMSON ST2030 ", 15) ||
+		    !strncmp(peer->useragent, "THOMSON ST2022 ", 15)) {
+			/* call-id@workaround.proformatique.com */
+			char callid[64];
+			char *stringp;
+
+			ast_copy_string(callid, p->callid, sizeof(callid));
+			stringp = callid;
+			strsep(&stringp, "@");
+			ast_string_field_build(p, callid, "%s@workaround.proformatique.com", callid);
+		}
 	} else {
 		/* Build temporary dialog for this message */
 		if (!(p = sip_alloc(NULL, NULL, 0, SIP_NOTIFY))) 
